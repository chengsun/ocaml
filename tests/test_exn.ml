exception Aaa
exception Bbb of int

let f x = if x > 0 then raise Aaa else 35
let g x = if x > 0 then raise (Bbb x) else 0

let () =
  try Printf.printf "before\n"; Printf.printf "f is %d\n" (f 30); with
  | Aaa -> Printf.printf "Aaa\n"
  | Bbb x -> Printf.printf "Bbb %d\n" x

let () =
  try Printf.printf "before\n"; Printf.printf "f is %d\n" (f (-30)); with
  | Aaa -> Printf.printf "Aaa\n"
  | Bbb x -> Printf.printf "Bbb %d\n" x

let () =
  try Printf.printf "before\n"; Printf.printf "g is %d\n" (g 30); with
  | Aaa -> Printf.printf "Aaa\n"
  | Bbb x -> Printf.printf "Bbb %d\n" x

let () =
  try Printf.printf "before\n"; Printf.printf "g is %d\n" (g (-30)); with
  | Aaa -> Printf.printf "Aaa\n"
  | Bbb x -> Printf.printf "Bbb %d\n" x
