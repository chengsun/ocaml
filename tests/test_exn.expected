before
Aaa
before
f is 35
before
Bbb 30
before
g is 0
